editor/scene_tree_dock.cpp

```diff
@@ -2890,6 +2905,32 @@ void SceneTreeDock::_clear_clipboard() {
 		memdelete(E->get());
 	}
 	node_clipboard.clear();
+	clipboard_resource_remap.clear();
+}
+
+void SceneTreeDock::_create_remap_for_node(Node *p_node, Map<RES, RES> &r_remap) {
+	List<PropertyInfo> props;
+	p_node->get_property_list(&props);
+
+	for (List<PropertyInfo>::Element *E = props.front(); E; E = E->next()) {
+		if (!(E->get().usage & PROPERTY_USAGE_STORAGE)) {
+			continue;
+		}
+
+		Variant v = p_node->get(E->get().name);
+		if (v.is_ref()) {
+			RES res = v;
+			if (res.is_valid()) {
+				if (res->get_path() == "" && !r_remap.has(res)) {
+					r_remap[res] = res->duplicate();
```

Comment:
You probably need to go recursive here, you don't want to deep duplicate because there may be external resources, but you might end up finding other resources from the original scene as children of this one that you might also want to remap.